include/cantera/thermo/ThermoPhase.h

```diff
@@ -1171,8 +1171,246 @@ class ThermoPhase : public Phase
+    void setEquivalenceRatio_Y(double phi, const compositionMap& fuelComp, const compositionMap& oxComp);
+    //@}
+ 
+    //! @name Compute Equivalence Ratio
+    //! @{
+    //! Compute the equivalence ratio for
+    //! the current mixture given fuel and oxidizer compositions.
+    //! Fuel and oxidizer compositions are specified by their mole fractions (do
+    //! not need to be normalized).
+    //! Elements C, S, H and O are considered for the oxidation. 
+    // 
+    //
+    /*!
+     * @param fuelComp   mole fractions of species in the fuel
+     * @param oxComp     mole fractions of species in the oxidizer 
+     * @param returns    equivalence ratio 
+     */
+    double getEquivalenceRatio_X(const double* fuelComp, const double* oxComp) const;
+    //! @copydoc ThermoPhase::getEquivalenceRatio_X(const double* fuelComp, const double* oxComp) 
+    double getEquivalenceRatio_X(const vector_fp& fuelComp, const vector_fp& oxComp) const;
+    //! @copydoc ThermoPhase::getEquivalenceRatio_X(const double* fuelComp, const double* oxComp) 
+    double getEquivalenceRatio_X(const std::string& fuelComp, const std::string& oxComp) const;
+    //! @copydoc ThermoPhase::getEquivalenceRatio_X(const double* fuelComp, const double* oxComp) 
+    double getEquivalenceRatio_X(const compositionMap& fuelComp, const compositionMap& oxComp) const;
+    
+    //! Compute the equivalence ratio for
+    //! the current mixture given fuel and oxidizer compositions.
+    //! Fuel and oxidizer compositions are specified by their mass fractions (do
+    //! not need to be normalized).
+    //! Elements C, S, H and O are considered for the oxidation. 
+    // 
+    //
+    /*!
+     * @param fuelComp   mass fractions of species in the fuel
+     * @param oxComp     mass fractions of species in the oxidizer 
+     * @param returns    equivalence ratio 
+     */
+    double getEquivalenceRatio_Y(const double* fuelComp, const double* oxComp) const;
+    //! @copydoc ThermoPhase::getEquivalenceRatio_Y(const double* fuelComp, const double* oxComp) 
+    double getEquivalenceRatio_Y(const vector_fp& fuelComp, const vector_fp& oxComp) const;
+    //! @copydoc ThermoPhase::getEquivalenceRatio_Y(const double* fuelComp, const double* oxComp) 
+    double getEquivalenceRatio_Y(const std::string& fuelComp, const std::string& oxComp) const;
+    //! @copydoc ThermoPhase::getEquivalenceRatio_Y(const double* fuelComp, const double* oxComp) 
+    double getEquivalenceRatio_Y(const compositionMap& fuelComp, const compositionMap& oxComp) const;
+ 
+    //! Compute the equivalence ratio for the current mixture from elemental
+    //! composition assuming that the fuel only contains the elements C, S, H
+    //! (and elements that do not oxidize) and the fuel only contains O (and
+    //! elements that do not oxidize)
+    // 
+    //
+    /*!
+     * @param returns    equivalence ratio 
+     */
+
+    double getEquivalenceRatio() const;
```

Comment:
So this is essentially an implementation of the previous definition of `getEquivalenceRatio`, is that correct? In that case, is it necessary or useful?

Reply:
This function is the special case of assuming that fuel and oxidizer are pure (what is sometimes called "local equivalence ratio", because it only considers the locally available oxygen and required oxygen and is therefore independent of fuel or oxidizer compositions). I introduced this function for convenience, because you can just write `gas.getEquivalenceRatio()` instead of specifying a concrete oxidizer and fuel composition like `gas.getEquivalenceRatio("H2:1","O2:1")`, where the actual values do not matter as long as the compositions only contain fuel and oxidizer elements, respectively. 
In the python interface, I solved this by defaulting the oxidizer and fuel compositions to `None`. However, defaulting the `compositionMap& oxComp` to `compositionMap{}` or `double* oxComp` to `nullptr` feels a but clunky in the C++ interface. I could rename the function to `getLocalEquivalenceRatio`, but this might be more confusing. What do you think?